### ISSUE TITLE
Undeploying bpfman selinux example programs hang on OpenShift

### ISSUE THREAD

**Issue by anfredette** [bpfman/bpfman-operator#331]
Running `make deploy-*-selinux` to an OpenShift cluster seems to work, but `make undeploy-*-selinux` hangs.

On closer inspection, after deploying, the selinux profile stays in the “Pending” state which doesn’t seem right.  Then, after running the undeploy command, the namespace is waiting on an selinux profile and a number of finalizers.

For more details search for the test case starting with "$ k apply -f go-tracepoint-counter-install-selinux.yaml" in [bpfman Selinux Test](https://gist.github.com/anfredette/6db1b43c77c39421baef69f9d662b760).

**Comment by Billy99** [comment]
There are two issues here:
> On closer inspection, after deploying, the selinux profile stays in the “Pending” state which doesn’t seem right.

This is because security-profiles-operator is currently deployed in OpenShift by making it a dependency of bpfman-operator via `config/manifests/dependencies.yaml`. As a result, the security-profiles-operator is deployed in the bpfman namespace. security-profiles-operator encounters issues with this because there are other daemonsets in the namespace.

See: https://github.com/kubernetes-sigs/security-profiles-operator/issues/2699

> Then, after running the undeploy command, the namespace is waiting on an selinux profile and a number of finalizers.

Upon further testing, if the application namespace is not deleted until after the SelinuxProfile is deleted, everything cleans up fine. This is a bug in security-profiles-operator.

See: https://github.com/kubernetes-sigs/security-profiles-operator/issues/2684

**Comment by Billy99** [comment]
This has been fixed in security-profiles-operator v0.9.1. Just waiting for it to get merged into OperatorHub:
[operator security-profiles-operator (0.9.1) #5983](https://github.com/k8s-operatorhub/community-operators/pull/5983)

Once that merges, the examples deployment code in bpfman repository will need to be updated:
[support SecurityProfilesOperator v0.9.1 #1500](https://github.com/bpfman/bpfman/pull/1500)
